bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsAccountHandler.java

```diff
@@ -0,0 +1,332 @@
+
+    private final Logger logger = LoggerFactory.getLogger(ElroConnectsAccountHandler.class);
+
+    private static final String ELRO_CLOUD_LOGIN_URL = "https://uaa-openapi.hekreu.me/login";
+    private static final String ELRO_CLOUD_URL = "https://user-openapi.hekreu.me/device";
+    private static final String ELRO_PID = "01288154146"; // ELRO Connects Enterprise PID on hekr cloud
+
+    private static final int TIMEOUT_MS = 2500;
+    private static final int INITIAL_DELAY_S = 5; // initial delay for polling to allow time for login and access token
+                                                  // retrieval to succeed
+    private static final int REFRESH_INTERVAL_S = 300;
+
+    private boolean enableBackgroundDiscovery;
+
+    private volatile @Nullable ScheduledFuture<?> pollingJob;
+    private final HttpClient client;
+
+    private Gson gson = new Gson();
+    private Type loginType = new TypeToken<Map<String, String>>() {
+    }.getType();
+    private Type deviceListType = new TypeToken<List<ElroConnectsConnector>>() {
+    }.getType();
+
+    private final Map<String, String> login = new HashMap<>();
+    private String loginJson = "";
+    private volatile @Nullable String accessToken;
+    private volatile Map<String, ElroConnectsConnector> devices = new HashMap<>();
+
+    public ElroConnectsAccountHandler(Bridge bridge, HttpClient client) {
+        super(bridge);
+        this.client = client;
+        login.put("pid", ELRO_PID);
+        login.put("clientType", "WEB");
+
+        // The ElroConnectsBridgeDiscoverySerivce will query for enableBackgroundDiscovery before the handler is
+        // initialized, therefore already set it here
+        ElroConnectsAccountConfiguration config = getConfigAs(ElroConnectsAccountConfiguration.class);
+        enableBackgroundDiscovery = config.enableBackgroundDiscovery;
+    }
+
+    @Override
+    public void initialize() {
+        ElroConnectsAccountConfiguration config = getConfigAs(ElroConnectsAccountConfiguration.class);
+        String username = config.username;
+        String password = config.password;
+        enableBackgroundDiscovery = config.enableBackgroundDiscovery;
+
+        if ((username == null) || username.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "@text/offline.no-username");
+            return;
+        }
+        if ((password == null) || password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "@text/offline.no-password");
+            return;
+        }
+
+        login.put("username", username);
+        login.put("password", password);
+        loginJson = gson.toJson(login);
+
+        // Do initial login to retrieve access token and start polling with small initial delay if background discovery
+        // is enabled), so the access token is available
+        scheduler.submit(this::login);
```

Comment:
Just question: Is there any reason to explicitly attempt login here instead of having the job perform this without any initial delay? If background discovery is disabled, what will/should happen if this one-time login fails (for example due to temporary network problem)?

Reply:
This one time login (background discovery disabled) will take the thing online when the login is successful, or offline with error when not. That will be visible in the UI as a timeout exception. So a communication error should be apparent. The way to recover would be to disable and enable the thing to restart initialization, no automatic recovery. If background discovery is enabled, it will try to login again on the next iteration, 5 min. later.
As the login and polling to the cloud service is only really required for discovery, by disabling background discovery the cloud call every 5 min. is avoided. Manually triggering a scan will just do one cycle of cloud communication.

Do you think I should automate a retry of the initial login when there is a network problem? I see that as separate from background discovery.

Reply:
Handled in [other comment](https://github.com/openhab/openhab-addons/pull/12653#issuecomment-1116174501), resolving this one.